### PR TITLE
Add descendent files to watched files

### LINKED ISSUE #1731: Recursive watch flag has no effect on imported files
Oddly enough the -r flag does not seem to make any difference on my system. I use the following command: 

`node-sass --watch development/style/main.sass production/bundle.css --recursive`

This process builds my file just fine, but it ignores any changes done in imported files. The imports work however as the changes to the imported files are included once I change my main.sass again...

Any ideas why this is?

General information:
- OS: Windows 10
- NPM version: 3.8.6
- Node version: 6.0.0
- Node Process: { http_parser: '2.7.0',  node: '6.0.0',  v8: '5.0.71.35',  uv: '1.9.0',  zlib: '1.2.8',  ares: '1.10.1-DEV',  icu: '56.1',  modules: '48',  openssl: '1.0.2g' }
- Node Platform: win32
- Node architecture: ia32
- node-sass version:
  node-sass       3.10.0  (Wrapper)       [JavaScript]
  libsass         3.3.6   (Sass Compiler) [C/C++]

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes #1731

### CODE REVIEW

**Comment by nschonni** [comment]
Thanks for the PR! Could you add a test in https://github.com/sass/node-sass/blob/master/test/cli.js

**Comment by marvinhagemeister** [comment]
@nschonni That's great! Thanks for having a look at the PR.

I'd love to write tests for this but I running the tests of `cli.js` alone takes more than **20s** for each run. 😞 

Haven't had any luck in getting watch mode to work either because of a failing binary detection. It defaults to some weird values:

```bash
/dev/github/node-sass/vendor/bar-foo-baz/binding.node  # <- bar-foo-baz?
```

File: bin/node-sass [hunk 1]
```diff
@@ -266,9 +266,18 @@ function watch(options, emitter) {
...
+    // descendents may be added, so we need a new graph
+    graph = buildGraph(options);
     graph.visitAncestors(file, function(parent) {
       files.push(parent);
     });
+
+    // Add children to watcher
+    graph.visitDescendents(file, function(child) {
+      gaze.add(child);
+      files.push(child);
```
Comment (xzyfer) [inline bin/node-sass]: You don't want to compile the decedents of the changed file. Please remove this.

**Comment by xzyfer** [comment]
You can use the `.only` method to run a subset of the test suite. i.e. `describe.only('...'` or `it.only('...`

File: bin/node-sass (hunk 1 above)
Comment (marvinhagemeister) [inline bin/node-sass]: Thanks for the feedback! You're right, that isn't necessary at all. Will fix that tomorrow

**Comment by nschonni** [comment]
@marvinhagemeister do you want to finish up the review, or maybe turn on collaborator contributions so it can get merged?

**Comment by marvinhagemeister** [comment]
@nschonni Sorry for not getting back any sooner, life got a bit hectic. Just updated the PR with the suggested change. Thanks for having another look.

**Comment by marvinhagemeister** [comment]
hold on something went wrong with the rebase

**Comment by xzyfer** [comment]
Thanks @marvinhagemeister!

**Comment by marvinhagemeister** [comment]
Awesome! Thanks for merging!

**Pull request merged**